src/context/Card-context/index.js

```diff
@@ -0,0 +1,157 @@
+import React, { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+
+export const CardContext  = React.createContext();
+
+const Provider = (props) => {
+  const [cardState, setCardState] = useState([
+    {
+      id: 1,
+      caption: 'One',
+      text: 'first.',
+      checked: false,
+      edited: false
+    },
+    {
+      id: 2,
+      caption: 'Two',
+      text: 'second',
+      checked: false,
+      edited: false
+    },
+    {
+      id: 3,
+      caption: 'Three',
+      text: 'third',
+      checked: false,
+      edited: false
+    },
+    {
+      id: 4,
+      caption: 'Four',
+      text: 'fourth',
+      checked: false,
+      edited: false
+    },
+    {
+      id: 5,
+      caption: 'Five',
+      text: 'Fifth',
+      checked: false,
+      edited: false
+    },
+    {
+      id: 6,
+      caption: 'Six',
+      text: 'sixth',
+      checked: false,
+      edited: false
+    },
+    {
+      id: 7,
+      caption: 'Seven',
+      text: 'seventh',
+      checked: false,
+      edited: false
+    },
+    {
+      id: 8,
+      caption: 'Caption',
+      text: 'Text...',
+      checked: false,
+      edited: false
+    }
+  ]);
+
+  function checkboxChange(id) {
+    const cardIndex = cardState.findIndex(c => {
+      return c.id === id;
+    });
```

Comment:
You can make it shorter:
```
 const cardIndex = cardState.findIndex(c => c.id === id);
```
The same could be applied everywhere in the code for similar cases.